src/test/java/com/redhat/pantheon/model/api/SlingResourceTest.java

```diff
@@ -0,0 +1,292 @@
+package com.redhat.pantheon.model.api;
+
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.testing.mock.sling.junit5.SlingContext;
+import org.apache.sling.testing.mock.sling.junit5.SlingContextExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Calendar;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith({SlingContextExtension.class, MockitoExtension.class})
+class SlingResourceTest {
+
+    private final SlingContext slingContext = new SlingContext();
+
+    @Test
+    public void fieldMapping() throws Exception {
+        slingContext.build()
+                .resource("/content/module1",
+                        "jcr:name", "my-name",
+                        "jcr:date", Calendar.getInstance(),
+                        "jcr:number", "10")
+                .commit();
+
+        TestResource model = new TestResource(slingContext.resourceResolver().getResource("/content/module1"));
+
+        assertEquals("my-name", model.NAME.get());
+        assertEquals(new Long(10), model.NUMBER.get());
+        assertNotNull(model.DATE.get());
+    }
+
+    @Test
+    public void childMapping() throws Exception {
+        // Given
+        slingContext.build()
+                .resource("/content/module1")
+                .resource("child")
+                .resource("grandchild")
+                .commit();
+
+        // When
+        TestResource model = new TestResource(slingContext.resourceResolver().getResource("/content/module1"));
+
+        // Then
+        model.CHILD
+                .get()
+                .GRANDCHILD
+                .get();
```

Comment:
Are we just testing to make sure this doesn't throw an exception? Is it possible that `GRANDCHILD.get()` might return null - would we need to test for that?

Reply:
Good catch. I've added an assertion (and actual code) to test for this scenario.